### PR TITLE
fix(shell): launch shell does not work with SSH VSCODE-327

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

Add proxy options to the current connection string to launch shell connected to a remote host, instead of localhost.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/telemetry/telemetryService.ts [hunk 1]
```diff
@@ -162,13 +157,9 @@ export default class TelemetryService {
...
-      const identity: AnalyticsNodeIdentity = {
-        anonymousId: this._segmentAnonymousId
-      };
-      if (this._segmentUserId) {
-        identity.userId = this._segmentUserId;
-      }
-      this._segmentAnalytics.identify(identity);
+      const segmentProperties = this.getTelemetryUserIdentity();
+      this._segmentAnalytics.identify(segmentProperties);
+      log.info('TELEMETRY identify', segmentProperties);
```
Comment (Anemy) [inline src/telemetry/telemetryService.ts:162]: These changes seem unrelated, is it drive by cleanup?

File: src/connectionController.ts [hunk 2]
```diff
@@ -713,6 +713,43 @@ export default class ConnectionController {
       : '';
   }
 
+  _getConnectionStringWithProxy(mongoClientConnectionOptions: { url: string; options: MongoClientOptions; }): string {
```
Comment (Anemy) [inline src/connectionController.ts:716]: Should we also use this method when we `Copy connection string`?
https://github.com/mongodb-js/vscode/blob/main/src/connectionController.ts#L725

File: src/telemetry/telemetryService.ts (hunk 1 above)
Comment (alenakhineika) [reply src/telemetry/telemetryService.ts:162]: Yeah clean up to make it closed to the recent fix in mongosh: https://github.com/mongodb-js/mongosh/pull/1229/files#diff-ac2009df67ba6eafe8b274126601bb9491e31f6920c72bbd1b1c33d34864e237R78-R86

File: src/connectionController.ts (hunk 2 above)
Comment (alenakhineika) [reply src/connectionController.ts:716]: We can copy a connection string for any connection on a sidebar, not only the active one. But proxy options are available only for the current connection, therefore we don't change the copy connection string behaviour.

**Review by Anemy — APPROVED** [review]
:shipit:
